File: 3.5 Dynamic Programming/Max 1D Range Sum/p10684.py
# Competitive Programming 3
# Problem 10684

while True:
    N = int(input())
    if N == 0: break

    count = 0
    seq = []
    while count < N:
        line = [int(x) for x in input().split()]
        count += len(line)
        seq += line

    curMax = 0;
    for i in range(N):
        curMax = max(curMax + seq[i], 0)
    
    if curMax > 0:
        print("The maximum winning streak is " + str(curMax) + ".")
    else:
        print("Losing streak.")
    
    
    

File: 3.5 Dynamic Programming/Max 1D Range Sum/p10684.cpp
// Competitive Programming 3
// Problem 10684

#include <iostream>
using namespace std;

int main(){
    while (1){
        int N;
        cin >> N;
        if (N == 0){
            break;
        }
        int curMax = 0, ans = 0;

        int num;
        while (N--){
            cin >> num;
            curMax = max(0, curMax + num);
            ans = max(ans, curMax);
        }

        if (ans == 0){
            cout << "Losing streak." << endl;
        }
        else{
            cout << "The maximum winning streak is " << ans << "."<< endl;
        }
    }

    return 0;
}
